src/frontend/src/handler/create_secret.rs

```diff
@@ -51,7 +52,15 @@ pub async fn handle_create_secret(
     }
     let with_options = WithOptions::try_from(stmt.with_properties.0.as_ref() as &[SqlOption])?;
 
-    let secret_payload = get_secret_payload(stmt.credential, with_options)?;
+    // Check for secret references in WITH options (forbid them during secret creation)
+    if !with_options.secret_ref().is_empty() {
+        return Err(ErrorCode::InvalidParameterValue(
+            "Secret references are not allowed when creating a secret".to_owned(),
```

Comment:
Hmm I suppose in the future we may need to support it and do recursion checks. Because `auth_token` is technically a `secret`.

Reply:
The fill_secrets interface is designed for resolving secrets at the same layer. I prefer to ban first the impl in following pr.